modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairAbstractFuture.java

```diff
@@ -261,10 +257,19 @@ else if (!canRemap)
 
             return;
         }
+        else
+            LSNR_CALLS_UPD.incrementAndGet(this);
 
-        for (GridPartitionedGetFuture<KeyCacheObject, EntryGetResult> fut : futs.values()) {
-            if (!fut.isDone() || fut.error() != null)
+        assert lsnrCalls <= futs.size();
+
+        if (isDone() || lsnrCalls != futs.size())
+            return;
+
+        synchronized (this) {
```

Comment:
I think it's possible to use `lsnrCalls` value immediately after incrementing it. This value is guarded from concurrency because incremented atomically, so we can leverage on this and remove any sync here. Am I missing smth here?